### DISCUSSION TITLE
Hey is it possible to make a custom transaction signing pop up in snaps?

### DISCUSSION THREAD

**Discussion by Arch0125** [MetaMask/snaps#1192] [Q&A]
I was looking to create a snap where the user can sign a msg/txs using BLS, is it possible for snaps to open up a separate pop-up for that ?

**Comment by ziad-saab** [comment]
In an upcoming Flask release, you'll be able to use `snap_dialog` to make custom dialogs. They won't look and feel like the MetaMask transaction window, but they'll allow you to get user input. You can read a draft of the documentation here:

https://github.com/MetaMask/metamask-docs/pull/607/files#diff-335fd934a17ae547f1db58a681cd5a2a70d03c03911f47d953ad22c77ba21692R450

You can already play with this feature if you build MetaMask Flask from source using the `develop` branch. Instructions are here:

https://github.com/MetaMask/metamask-extension#building-locally

Make sure to be on the develop branch and use `--build-type=flask` to get the developer preview

**Reply by Montoya** [reply]
To add, you can also download one of the builds here: 

https://github.com/MetaMask/metamask-extension/releases/tag/v10.25.0

(scroll down to Assets and choose MetaMask-Flask-Chrome or MetaMask-Flask-Firefox

Then you can install it into your browser as long as you have developer mode enabled. For Google Chrome: 

1. Go to chrome://extensions/ and check the box for Developer mode in the top right.
2. Drag and drop the zip file you downloaded from the releases page into the Chrome window

Just make sure you uninstall or disable any other versions of MetaMask before installing this.

**Comment by Arch0125** [comment]
Is there an alternative for this 

```
async function getPubKey() {
  const PRIV_KEY = await snap.request({
    method: 'snap_getAppKey',
  });
  return bls.getPublicKey(PRIV_KEY);
} 
```

as this method doesnt seem to work as of now

**Reply by ritave** [reply]
Hey @Arch0125, we have replaced `snap_getAppKey` with `snap_entropy` in Flask 10.25.

* See all the breaking changes [here](https://github.com/MetaMask/snaps-monorepo/discussions/1198).
* See the documentation for `snap_entropy` [here](https://docs.metamask.io/guide/snaps-rpc-api.html#snap-getentropy).
